api/tests/certification/course/integration/infrastructure/repositories/sco-certification-candidate-repository_test.js

```diff
@@ -323,60 +259,111 @@ describe('Certification | Course | Integration | Repository | SCOCertificationCa
+          ...candidate,
+          sessionId: sessionPublishedId,
+        });
+        databaseBuilder.factory.buildCertificationCourse({
+          sessionId: unpublishedSessionId,
+          lastName: candidateFromUnpublishedSession.lastName,
+          firstName: candidateFromUnpublishedSession.firstName,
+          isPublished: false,
+          userId: candidateFromUnpublishedSession.userId,
+        });
+        databaseBuilder.factory.buildCertificationCourse({
+          sessionId: sessionPublishedId,
+          lastName: candidateFromPublishedSession.lastName,
+          firstName: candidateFromPublishedSession.firstName,
+          isPublished: true,
+          userId: candidateFromPublishedSession.userId,
+        });
 
-      const sessionIdOne = databaseBuilder.factory.buildSession({ publishedAt: '2024-02-01' }).id;
-      const sessionIdTwo = databaseBuilder.factory.buildSession({ publishedAt: '2024-01-01' }).id;
+        await databaseBuilder.commit();
 
-      const candidateLinkedToTheFirstSession = databaseBuilder.factory.buildCertificationCandidate({
-        ...candidate,
-        sessionId: sessionIdOne,
-      });
-      databaseBuilder.factory.buildCertificationCourse({
-        createdAt: firstCertificationCourseStartDate,
-        sessionId: candidateLinkedToTheFirstSession.sessionId,
-        lastName: candidateLinkedToTheFirstSession.lastName,
-        firstName: candidateLinkedToTheFirstSession.firstName,
-        isPublished: true,
-        userId: candidateLinkedToTheFirstSession.userId,
-        pixCertificationStatus: 'rejected',
-      });
-      const candidateLinkedToTheSecondSession = databaseBuilder.factory.buildCertificationCandidate({
-        ...candidate,
-        sessionId: sessionIdTwo,
-      });
-      databaseBuilder.factory.buildCertificationCourse({
-        createdAt: secondCertificationCourseStartDate,
-        sessionId: candidateLinkedToTheSecondSession.sessionId,
-        lastName: candidateLinkedToTheSecondSession.lastName,
-        firstName: candidateLinkedToTheSecondSession.firstName,
-        isPublished: true,
-        userId: candidateLinkedToTheSecondSession.userId,
-        pixCertificationStatus: 'validated',
-      });
-      await databaseBuilder.commit();
+        // when
+        const candidatesIds = await scoCertificationCandidateRepository.findIdsByOrganizationIdAndDivision({
+          organizationId,
+          division,
+        });
 
-      // when
-      const candidatesIds = await scoCertificationCandidateRepository.findIdsByOrganizationIdAndDivision({
-        organizationId,
-        division,
+        // then
+        expect(candidatesIds).to.deep.equal([candidateFromPublishedSession.id]);
       });
 
-      // then
-      expect(candidatesIds).to.deep.equal([candidateLinkedToTheSecondSession.id]);
+      it('should retrieve the latest candidate when one has entered multiple sessions', async function () {
```

Comment:
on peut splitter en deux:
```suggestion
      context('when one candidate has entered multiple sessions', async function () {
        it('should retrieve the latest', async function () {
```

Reply:
done :) (le contexte existait déjà et englobait ce test. J'ai enlevé la mention superflue)